ssz/src/main/java/tech/pegasys/teku/ssz/SszMutableList.java

```diff
@@ -28,6 +30,31 @@
   @Override
   SszList<SszElementT> commitChanges();
 
+  /**
+   * Set the same value across the specified indices
+   *
+   * @param value The value to set
+   * @param startIndex The first index in the range to be modified (inclusive)
+   * @param endIndex The last index in the range to be modified (exclusive)
+   */
+  default void setAll(SszElementT value, int startIndex, int endIndex) {
+    checkArgument(
+        startIndex <= endIndex, "Start index must be less than or equal to the end index");
+    for (int i = startIndex; i < endIndex; i++) {
+      set(i, value);
+    }
+  }
+
+  /**
+   * Set the same value from index zero (inclusive) up to {@code endIndex} (exclusive).
+   *
+   * @param value The value to set
+   * @param endIndex The index defining the end of the range to set (exclusive)
+   */
+  default void setAll(SszElementT value, int endIndex) {
+    setAll(value, 0, endIndex);
+  }
+
```

Comment:
(nit) If the same methods would be needed for `SszMutableVector`  it makes sense to add `SszMutableCollection` and move methods there